src/tls/openssl/tls.c

```diff
@@ -182,6 +182,10 @@ int tls_alloc(struct tls **tlsp, enum tls_method method, const char *keyfile,
 			goto out;
 		}
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+		SSL_CTX_set_verify(tls->ctx, SSL_VERIFY_PEER, NULL);
+#endif
+
```

Comment:
regarding this diff I have two comments:

1. one would think that the default behaviour for openssl clients is to
  verify the TLS Server ? If not it makes the whole certificate stuff useless.
  perhaps the code is needed, but do you have references to relevant docs ?

2. the code block is inside a block (`if (keyfile)`) that is only executed if the
  certificate parameter is specified. This is typically done for TLS servers, but
  I think your usecase was a TLS client ? perhaps the code block should be moved
  to a place that is always executed for both client/server ?